Update search result styling

i0sea opened the pull request:
#### Summary of changes
**Asana Ticket:** [Update search result styling](https://app.asana.com/0/555089885850811/1201244628068090/f)
Make some styling changes to search results

- Add project icon for results for projects
- Update placement of mode/line icons to be next to station/stop or project icon
- Update placement of text on to second line if more than two icons

thecristen reviewed (changes requested):
I'm seeing all the icons on the left! 👍🏼 A few requested updates:

* The text of the result is supposed to be bumped to a new line if there are more than two icons with it, I'm not seeing it here:

![image](https://user-images.githubusercontent.com/2136286/148565010-4d0647f5-4649-48ea-a597-8bbb2b5c594f.png)

* I'm not seeing a project icon here? It's a little hammer icon (example in mock 2.1.1.3)
* There are some cases where it looks like we might need to add a space between the icon and text, example in two spots here. Can we get the icon spacing consistent?

![image](https://user-images.githubusercontent.com/2136286/148565357-a123d54e-1fe2-4eed-b01a-62ce9d8b4d77.png)

i0sea commented:
What are some examples of projects that should have a project icon?

i0sea commented:
<img width="883" alt="Screen Shot 2022-01-14 at 11 52 39 AM" src="https://user-images.githubusercontent.com/91911166/149578716-e502a0f4-688a-42df-a739-72471a2e0e61.png">
updated text to be on second line but now the icons are underlined on hover. Moved the icons span to be outside the anchor tag which got rid of the underlines but moved everything back to one line. Open to suggestions on a fix to resolve both issues.

thecristen commented on apps/site/assets/js/algolia-result.js:
```diff
@@ -30,17 +30,23 @@ export const TEMPLATES = {
     {{^id}}
     <a class="${SELECTORS.result} c-search-result__link u-no-underline" href="{{hitUrl}}" data-queryid="{{analyticsData.queryID}}" data-hit-position="{{analyticsData.position}}" data-objectid="{{analyticsData.objectID}}">
     {{/id}}
+      <span>{{{hitIcon}}}</span>
+      <span class="c-search-result__feature-icons">
+      {{#hitFeatureIcons}}
+        {{{.}}}
+      {{/hitFeatureIcons}}
+      <br/>
+    </span>
```
Would it work to put these lines immediately before the <a> on line 41? I'm thinking that taking them outside the link might help with the underlining issue.

Otherwise I think it makes sense to file a new ticket for that formatting and let that part go.

thecristen commented:
> <img alt="Screen Shot 2022-01-14 at 11 52 39 AM" width="883" src="https://user-images.githubusercontent.com/91911166/149578716-e502a0f4-688a-42df-a739-72471a2e0e61.png">
> 
> updated text to be on second line but now the icons are underlined on hover. Moved the icons span to be outside the anchor tag which got rid of the underlines but moved everything back to one line. Open to suggestions on a fix to resolve both issues.

Looked at the code and the link has a `.u-no-underline` class on it... it's not even supposed to be underlined in the first place. 😩 So thanks for catching that! Can you file a ticket to remove this underlining?

thecristen commented:
Bypassing the unrelated failing test to merge in, let's goooo
